[25기 정찬영] Westagram-1팀 작업

gkongs opened the pull request:
## 수정 사항 간략한 한줄 요약
**Add: 기존 React 작업 사항 추가 및 공용 파일 수정**

images/chanyoung |  이미지 추가
common.scss | reset import 추가
reset.scss | 불필요한 주석 제거
.prettier | 한 줄 길이 설정

## 수정 사항들 자세한 내용
필요한 이미지 추가
한 줄 200자로 설정
common.scss만 사용하도록 해당 파일에 reset import 추가

## 기타 질문 및 특이 사항
공용파일 수정은 따로 브런치 생성해서 하는지 모르고 했었지만 앞으론 브런치 하나 생성하여 공용파일 내용 작업하고 푸쉬하겠습니다!

두번째 'Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현' 커밋은 Mission 2 입니다
기능 구현하면서 컴포넌트화를 같이해서 3-4 Mission이 함께 입니다
자꾸 커밋 메시지를 잘못 작성한 걸 올리네요..  항상 다시 한번 확인하고 올리는 습관을 들이겠습니다
(덮어쓰기해서 바꿀순 있다는데 무서워서 안했습니다..ㅠ)


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [ ] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [ ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ ] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

gkongs committed:
Add: 기존 React 작업 사항 추가 및 공용 파일 수정
common.scss | reset import 추가
reset.scss | 불필요한 주석 제거
.prettier | 한 줄 길이 설정

gkongs committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현

gkongs committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현
- components 폴더 추가
- LoginForm, Id, Pw, Button 컴포넌트화

gkongs committed:
Fix: 폴더 변경
Login->components->Login에서 Login->components로 변경

gkongs committed:
Add : Mission 3,4 - 댓글 기능 + 댓글 컴포넌트화, props로 데이터 전달 구현
댓글 삭제 기능 추가
댓글 좋아요 아이콘 삭제 및 삭제 아이콘 위치 변경
Please enter the commit message for your changes. Lines starting

gkongs committed:
Refactor: scss nesting + if문 변경
- Login, Main. scss nesting
- short-circuit evaluation 적용

gkongs committed:
Add, Refactor: Mock + Login 리팩토링
- Mock 과제를 위한 json 파일 추가
- Feed 컴포넌트 추가
- Login Nesting 관련 className 변경
- css -> scss 변경시 빠진 style 요소 추가 ( header - font / input - outline:none )

gkongs committed:
Refactor: 셀프 리팩토링 50% 진행
- Main 파일 리팩토링 진행중
- Main scss nesting 관련 이름 변경 작업 진행중
- console.log 삭제
- import 위치 변경

sangcheol-LEE commented on src/pages/chanyoung/Login/components/Id.js:
```diff
@@ -0,0 +1,21 @@
+  checkVaild = e => {
+    e.target.value.includes('@')
+      ? this.props.changeIsVaild('id', true)
+      : this.props.changeIsVaild('id', false);
```
구조분해할당 가도 되고 안하셔두되고 ㅎㅎ...

gkongs replied:
이 부분 비구조화 해봤는데, 안했을 때가 코드 줄도 적고 가독성이 더 좋은 것 같아서 이 부분은 수정하지 않겠습니다!

gkongs committed:
Refactor: Peer Review 후 개선점 적용

- scss name 형식 통일 (낙타)
- scss nesting
- img tag에 alt 속성 값 추가

gkongs committed:
Refactor: 추가 리팩토링 작업

- 함수 순서 변경
- 비구조화 & 비구조화 삭제
- 사용하지 않는 파라메터 삭제
- console.log 삭제
- state name 변경

saengmotmi commented on src/pages/chanyoung/Login/Login.js:
```diff
@@ -1,9 +1,38 @@
+import './Login.scss';
 
 class Login extends React.Component {
+  goToMain = (id, pw, e) => {
```
data fetching 하는 함수와 경로 이동하는 함수를 분리하는게 좋을거 같아요. 하나의 함수가 너무 여러가지 일을 하는거 같습니다

gkongs replied:
리뷰 확인 후에 부모 컴포넌트(Login)에 굳이 id, pw, event 객체를 보낼 필요가 없다고 생각해서 LoginForm에 함수를 하나 만들어서 주는 방식으로 했는데, 이런식으로 하는게 맞을지 아니면 이걸 더 분할하라는 말씀이신지 잘 모르겠습니다 ㅠㅠ
```js
// LoginForm.js
checkUser = e => {
    e.preventDefault();
    fetch('http://10.58.2.15:8000/login', {
      method: 'POST',
      body: JSON.stringify({
        email: this.state.id.date,
        password: this.state.pw.date,
      }),
    })
      .then(res => res.json())
      .then(result => {
        result.MESSAGE === 'SUCCESS' && this.props.goToMain();
      });
  };
```
```js
// Login.js
goToMain = () => this.props.history.push('/Main-ChanYoung');
```

saengmotmi replied:
이렇게 하시면 될거 같아요!

saengmotmi commented on src/pages/chanyoung/Login/Login.scss:
```diff
@@ -0,0 +1,84 @@
+    margin-bottom: 25px;
+  }
+
+  .button {
+    border: none;
+    background-color: blue;
+    color: white;
+    font-size: 25px;
+  }
+
+  .buttonDisabled {
+    border: none;
+    background-color: #c4e1fb;
+    color: white;
+    font-size: 25px;
+  }
```
동일한 속성이 반복되고 있는데 button에 대한 스타일과 disabled에 대한 스타일을 별도 셀렉터로 분리하면 중복을 줄일 수 있겠네요

gkongs replied:
mixin 인자로 background-color를 주는 방식으로 변경 했습니다!

saengmotmi commented on src/pages/chanyoung/Login/components/Id.js:
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+class Id extends React.Component {
+  checkVaild = ({ target: { value } }) => {
+    value.includes('@')
+      ? this.props.changeIsVaild('id', value, true)
+      : this.props.changeIsVaild('id', value, false);
+  };
```
```suggestion
  checkVaild = (e) => {
    this.props.changeIsVaild('id', e.target.value)
  };
```
- value.includes('@')가 이미 boolean 입니다
- 코드를 정리하다 보면 2번 째 인자와 3번째 인자가 모두 value 값에 의존하고 있는데, 이미 value 전체를 넘겨주고 있으므로 T/F 판별은 changeIsValid 내에서 처리하면 될거 같습니다
- 그리고 구조분해 할당은 1 depth 정도가 충분한거 같아요

gkongs replied:
완료했습니다

saengmotmi commented on src/pages/chanyoung/Login/components/LoginForm.js:
```diff
@@ -0,0 +1,45 @@
+      <form
+        className="login"
+        onSubmit={e => {
+          this.props.goToMain(id.data, pw.data, isBtn, e);
```
```suggestion
          this.props.goToMain(e, id, pw);
```
위 리뷰와 마찬가지로 id, pw를 이미 넘겨줄 수 있기 때문에 isBtn은 불필요합니다

gkongs replied:
이건 실수로 지우지 않았었네요.. 수정했습니다!

saengmotmi commented on src/pages/chanyoung/Main/Main.js:
```diff
@@ -1,8 +1,181 @@
+                  <span className="storiesUserTime">20시간 전</span>
+                </div>
+              </div>
+              <div className="storiesUser">
```
반복되는 요소는 컴포넌트화 해주세요

gkongs replied:
완료했습니다

saengmotmi commented on src/pages/chanyoung/Main/Main.scss:
```diff
@@ -0,0 +1,333 @@
+@import '../../../styles/common.scss';
+
+:root {
+  --border: 1px solid rgba(0, 0, 0, 0.2);
+  --background-color--gray: #fafafa;
+}
```
common으로 가는게 어떨까요?

saengmotmi commented on src/pages/chanyoung/Main/components/feeds/Comment.js:
```diff
@@ -0,0 +1,76 @@
+      comment: e.target[0].value,
+    };
+
+    this.setState(({ commentsInfo: { list } }) => {
+      return {
+        commentsInfo: {
+          list: [...list, newComment],
+          nextKey: nextKey + 1,
+        },
+      };
+    });
```
```suggestion
    this.setState(prevState => {
      const { list } = prevState.commentsInfo
      return {
        commentsInfo: {
          list: [...list, newComment],
          nextKey: nextKey + 1,
        },
      };
    });
```
구조분해 할당 Depth가 너무 깊어져서 알아보기 어려운거 같아요

saengmotmi commented on src/pages/chanyoung/Main/components/feeds/Comment.js:
```diff
@@ -0,0 +1,76 @@
+  deleteComment = key => {
+    this.setState(({ commentsInfo: { list, nextKey } }) => ({
+      commentsInfo: {
+        list: list.filter(e => key !== e.key),
```
e 보다 더 구체적인 이름이 있을거 같습니다

gkongs replied:
e -> comment

saengmotmi commented on src/pages/chanyoung/Main/components/feeds/Comment.js:
```diff
@@ -0,0 +1,76 @@
+    this.setState(({ commentsInfo: { list, nextKey } }) => ({
+      commentsInfo: {
+        list: list.filter(e => key !== e.key),
+        nextKey: nextKey,
```
```suggestion
        nextKey
```
key와 value가 이름이 같으면 축약 가능합니다

gkongs replied:
놓쳤던 부분이네요 수정했습니다!

saengmotmi commented on src/pages/chanyoung/Main/components/feeds/CommentForm.js:
```diff
@@ -0,0 +1,14 @@
+class CommentForm extends React.Component {
+  render() {
+    return (
+      <form className="form" onSubmit={e => this.props.addCommnet(e)}>
```
```suggestion
      <form className="form" onSubmit={this.props.addCommnet}>
```
함수를 그대로 넘겨줘도 동일한 결과입니다

gkongs replied:
넵 수정했습니다

saengmotmi commented on src/pages/chanyoung/Main/components/feeds/CommentWnd.js:
```diff
@@ -0,0 +1,28 @@
+import { faBackspace } from '@fortawesome/free-solid-svg-icons';
+
+class CommentWnd extends React.Component {
+  showComment = () => {
```
메서드로 분리해줘도 되지만 render 안쪽에 바로 적어주는게 조금 더 좋아보이네요

gkongs replied:
수정했습니다

saengmotmi commented on src/pages/chanyoung/Main/components/feeds/CommentWnd.js:
```diff
@@ -0,0 +1,28 @@
+  showComment = () => {
+    const { commentsInfo, deleteComment } = this.props;
+
+    return commentsInfo.list.map(e => (
```
여기도 e 같은 이름 피해주세요 data, info 등의 추상적인 이름 역시 마찬가지입니다

gkongs replied:
제가 생각하기엔 댓글 정보에서 댓글 리스트를 가져온 값이니까 e 는 comment가 맞는 것 같아서 comment로 수정하였습니다 🚀

saengmotmi reviewed (commented):
리뷰 확인해주세요!

saengmotmi commented:
@gkongs git commit --amend 커맨드로 커밋 메시지 수정할 수 있습니다

gkongs committed:
Fix: 멘토 리뷰 적용

=== Login ===

- goToMain 함수 분할, 필요하지않은 state 끌어올리기 삭제
- 중복 스타일링 삭제( mixin 사용)
- checkVaild 삭제, changeIsVaild에서 모두 처리하도록 변경
- 불필요한 파라메터 삭제

=== Main ===

- Stories 컴포넌트화
- 중복되는 StoriesUser 부분 컴포넌트화 후 해당  데이터 추가
- showComment 삭제 후 해당 내용 렌더에 직접 작성
- 추상적인 이름 변경 및 리팩토링
